src/lib/redux/customerDataSlice.ts

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
+import { CustomerRecord, SiteId } from '../airtable/interface';
+import { setCurrSite } from './siteDataSlice';
+
+export const EMPTY_CUSTOMER: CustomerRecord = {
+    id: '',
+    name: '',
+    meterNumber: 0,
+    tariffPlanId: '',
+    isactive: false,
+    hasmeter: false,
+    outstandingBalance: '',
+    meterReadingIds: [],
+    meterReadings: [],
+    paymentIds: [],
+    payments: [],
+    customerUpdateIds: [],
+    customerUpdates: [],
+    totalAmountBilledfromInvoices: 0,
+    totalAmountPaidfromPayments: 0,
+}
+
+// TODO: @julianrkung, We should eventually make this a Record<SiteId, Record<CustomerId, CustomerRecord>> map for efficiency
+// We can't do this yet because customers created while offline do not receive an id yet
```

Comment:
Non-blocking, but is this actually solvable (aren't the IDs generated by Airtable - hence the create call needs to go through to get the customer's ID at all?)

Reply:
We're deciding what to do here. I was tossing around the idea of having temporary negative IDs for customers created offline (once the customer is created, we'll pull from airtable and override with the real ID). I think there's a need to be able to uniquely identify customers created even while offline, and I don't know a better solution.